envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -18,7 +18,28 @@ option (gogoproto.equal_all) = true;
 
 // Upstream host identifier.
 message Endpoint {
+  // The upstream host address.
   core.Address address = 1;
+
+  // [#not-implemented-hide:] The optional health check configuration.
+  message HealthCheckConfig {
+    // Optional alternative health check port value.
+    //
+    // By default the health check address port of an upstream host is the same
+    // as the host's serving address port. This provides an alternative health
+    // check port. If set it allows an upstream host to have different
+    // health check address port.
+    uint32 port_value = 1;
```

Comment:
I assume 0 means no alt port? If so can we document that?

Reply:
Try to address this in 75662e3. Please let me know if that one is either good enough or need improvement. Thanks.